R/cql-geom-predicates.R

```diff
@@ -105,17 +105,7 @@ cql_geom_predicate_list <- function() {
 
 sf_text <- function(x) {
 
-  if (!inherits(x, c("sf", "sfc", "sfg", "bbox"))) {
-    stop(paste(deparse(substitute(x)), "is not a valid sf object"),
-         call. = FALSE)
-  }
-
-  ## If too big here, drawing bounding
-  if (utils::object.size(x) > getOption("bcdata.max_geom_pred_size", 5E5)) {
-    warning("The object is too large to perform exact spatial operations using bcdata.
-             To simplify the polygon, a bounding box was drawn around the polygon and all
-             features within the box will be returned. Options include further processing
-             with on the returned object or simplify the object.", call. = FALSE)
+  if (bcdc_size_check(x)) {
```

Comment:
I think when `bcdc_size_check()` is called inside this function, an additional message/warning should be emitted saying that it was converted to a bbox. Otherwise the message/warning from `bcdc_size_check()` doesn't tell the user what happened, just that it was too big.